### PR TITLE
Slicing program studi detail page for Sarjana and Diploma Sistem Informasi

### PR DESCRIPTION
**Pull request by Rifqi9604**
Successfully slicing two detail pages for Program Studi

### CODE REVIEW

File: src/components/ui/accordion.tsx
Comment (nugrahaAdhitama) [inline src/components/ui/accordion.tsx:1]: Apa bedanya accordion ini sama accordion `src/components/ui/Accordion/Accordion.tsx`?

Comment (Rifqi9604) [reply src/components/ui/accordion.tsx:1]: Gw perjelas lagi, accordion itu dari shadcn, nah Accordion.tsx itu custom component yang di edit dari accordion itu

Comment (nugrahaAdhitama) [reply src/components/ui/accordion.tsx:1]: ok

File: src/components/ui/Accordion/Accordion.tsx
Comment (nugrahaAdhitama) [inline src/components/ui/Accordion/Accordion.tsx:1]: Ini dari shadcn juga kan?

File: src/components/ProgramStudi/FAQs/FAQ.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    content:
+      "Yes. It comes with default styles that match the other components&apos; aesthetic.",
+  },
+  {
+    id: 3,
+    title: "Capaian Pelajaran UPNVJ",
+    content:
+      "Yes. It comes with default styles that match the other components&apos; aesthetic.",
+  },
+];
```
Comment (nugrahaAdhitama) [inline src/components/ProgramStudi/FAQs/FAQ.tsx:29]: Pisahin ke file lain

**Review by nugrahaAdhitama — CHANGES_REQUESTED** [review]
## Code Review Summary

This review covers three components: `PeminatanCard`, `AccordionProgramStudi`, and `ProfilLulusan`. Each component is evaluated on general observations, specific issues, and suggestions for improvement. The focus is on enhancing readability, maintainability, and adherence to SOLID principles.

---

### 1. **Component: `PeminatanCard` (`src/components/ui/PeminatanCard/PeminatanCard.tsx`)**

#### **General Observations**
- The component is well-structured but can be improved for better readability and maintainability.
- Inline styles and repetitive class names create redundancy.
- The use of `index` as a key in the `map` function may lead to potential bugs.
- The component would benefit from separating concerns by extracting repeated JSX into smaller components.

#### **Specific Feedback**
- **Scope 1: Use of Index as Key in List Rendering**
  - **Issue**: Using `index` as a key can cause issues when list items are reordered, added, or removed.
  - **Suggestion**: Use a unique identifier from the `peminatan` object (e.g., `peminatan.id`) as the key.

- **Scope 2: Inline JSX in Map Function**
  - **Issue**: Complex and repetitive JSX inside the `map` function reduces readability.
  - **Suggestion**: Extract the JSX block into a separate component, such as `PeminatanCardItem`.

- **Scope 3: ClassName Redundancy**
  - **Issue**: Repetitive and long class names make the component harder to maintain.
  - **Suggestion**: Create reusable CSS classes or use CSS modules.

- **Scope 4: Component Composition**
  - **Issue**: The component handles both layout and rendering of card elements, making it harder to maintain.
  - **Suggestion**: Break down the component into smaller components, separating layout from card rendering logic.

#### **Conclusion**
The `PeminatanCard` component is well-structured but can be improved by following best practices. Addressing the use of indexes as keys, reducing inline JSX complexity, and managing className redundancy will make the component more maintainable and scalable.

---

### 2. **Component: `AccordionProgramStudi` (`src/components/ui/Accordion/AccordionProgramStudi.tsx`)**

#### **General Observations**
- The component is fairly well-structured, but readability and maintainability can be improved.
- The tight coupling of data and UI makes the component less flexible.
- Inline styles and class names can be refactored for better maintainability.
- The component could better adhere to SOLID principles.

#### **Specific Feedback**
- **Scope 1: Tight Coupling of Data and UI**
  - **Issue**: `accordionDatas` is declared within the UI logic, leading to tight coupling.
  - **Suggestion**: Extract `accordionDatas` to a separate file or pass it as a prop.

- **Scope 2: Class Name Redundancy**
  - **Issue**: Repetitive inline classes make the component harder to maintain.
  - **Suggestion**: Use CSS modules, styled-components, or utility classes.

- **Scope 3: Component Flexibility**
  - **Issue**: The component is tightly bound to specific data structures and styling.
  - **Suggestion**: Increase flexibility by accepting more props for customization.

- **Scope 4: Repetitive Code in `AccordionItem`**
  - **Issue**: Repetitive JSX in `AccordionItem` leads to a larger codebase.
  - **Suggestion**: Abstract the `AccordionItem` creation into a separate function or component.

#### **Conclusion**
The `AccordionProgramStudi` component can be enhanced by decoupling data from the UI, reducing class name redundancy, and increasing flexibility. These changes will improve maintainability and adherence to SOLID principles.

---

### 3. **Component: `ProfilLulusan` (`src/components/ProgramStudi/ProfilLulusan/ProfilLulusan.tsx`)**

#### **General Observations**
- The component is well-structured but can be optimized for readability and maintainability.
- Tightly coupled data and UI logic reduce flexibility and reusability.
- Repetitive class names could be abstracted for maintainability.
- The component could better follow SOLID principles.

#### **Specific Feedback**
- **Scope 1: Data and UI Coupling**
  - **Issue**: `graduateProfiles` data is defined within the same file as the UI logic.
  - **Suggestion**: Move `graduateProfiles` to a separate file or pass it as a prop.

- **Scope 2: Repetitive Class Names**
  - **Issue**: Repetitive class names make the component harder to maintain.
  - **Suggestion**: Create reusable CSS classes or use CSS modules.

- **Scope 3: Component Decomposition**
  - **Issue**: The component mixes layout, data mapping, and content presentation logic.
  - **Suggestion**: Break the component into smaller, focused components, like `GraduateProfileCard`.

- **Scope 4: Inline Styles and Magic Numbers**
  - **Issue**: Specific pixel values reduce flexibility and maintainability.
  - **Suggestion**: Replace with generic classes or theme variables.

#### **Conclusion**
The `ProfilLulusan` component is functional but can benefit from improvements in maintainability, scalability, and adherence to SOLID principles. Decoupling data from UI, reducing class name redundancy, and breaking the component into smaller parts will create a more modular and testable codebase.

---

File: src/components/ui/Accordion/Accordion.tsx
Comment (Rifqi9604) [reply src/components/ui/Accordion/Accordion.tsx:1]: Iya, tadinhynya ini yang accordion.tsx, gw rename jadi Accordion.tsx, terus gw masukin ke satu folder ada 2 file, Accordion.tsx yang dari shadcn sama AccordionProgramStudi.tsx

File: src/components/ProgramStudi/FAQs/FAQ.tsx (hunk 1 above)
Comment (Rifqi9604) [reply src/components/ProgramStudi/FAQs/FAQ.tsx:29]: Oke noted

**Review by nugrahaAdhitama — APPROVED** [review]
Approved

**Pull request merged**